pages/datasets/[datasetid]/access/research-purpose.vue

```diff
@@ -30,31 +32,134 @@ const generateCombinedFullName = (name: string) => {
   trigger: ["input", "blur"],
   validator() {
-    if (!researchPurpose.value?.trim()) {
+    if (!agreementFormState.value.research_purpose?.trim()) {
       return new Error("May not be blank.");
     }
 
     if (!validResearchPurpose.value) {
       return new Error(
-        `Must be between ${minLength} and ${maxLength} characters.`,
+        `Must be between ${minResearchPurposeLength} and ${maxResearchPurposeLength} characters.`,
       );
     }
   },
 };
+const defaultValidationMessage = `Required. May be at most ${defaultMaxLength} characters.`;
+const nonDiabetesRules = {
+  collaborator_data_access: {
+    message: "This field is required.",
+    required: true,
+    trigger: ["change", "blur"],
+    type: "boolean",
+  },
+  research_project_title: {
+    max: defaultMaxLength,
+    message: defaultValidationMessage,
+    required: true,
+    trigger: ["input", "blur"],
+  },
+  signing_official_address: {
+    max: defaultMaxLength,
+    message: defaultValidationMessage,
+    required: true,
+    trigger: ["input", "blur"],
+  },
+  signing_official_email: {
+    message: "Required. Must be a valid email address.",
+    required: true,
+    trigger: ["input", "blur"],
+    type: "email",
+  },
+  signing_official_first_name: {
+    max: defaultMaxLength,
+    message: defaultValidationMessage,
+    required: true,
+    trigger: ["input", "blur"],
+  },
+  signing_official_institution: {
+    max: defaultMaxLength,
+    message: defaultValidationMessage,
+    required: true,
+    trigger: ["input", "blur"],
+  },
+  signing_official_last_name: {
+    max: defaultMaxLength,
+    message: defaultValidationMessage,
+    required: true,
+    trigger: ["input", "blur"],
+  },
+  signing_official_phone_number: {
+    max: defaultMaxLength,
+    message: defaultValidationMessage,
+    required: true,
+    trigger: ["input", "blur"],
+  },
+  signing_official_title: {
+    max: defaultMaxLength,
+    message: defaultValidationMessage,
+    required: true,
+    trigger: ["input", "blur"],
+  },
+  study_overview: {
+    max: maxOverviewLength,
+    message: `Required. Must be between ${minOverviewLength} and ${maxOverviewLength} characters.`,
+    min: minOverviewLength,
+    required: true,
+    trigger: ["input", "blur"],
+  },
+};
+const validDacFormState = computed(() => {
+  if (!nonDiabetesResearch.value) {
+    return true;
+  }
+
+  const allValid = Object.keys(nonDiabetesRules)
+    .map((key) => {
+      const value = agreementFormState.value[key];
+      if (value === undefined || value === null) {
+        return false;
+      }
+
+      const min = nonDiabetesRules[key].min;
+      const max = nonDiabetesRules[key].max;
+
+      if (min || max) {
+        const tooLong = max ? value.trim().length > max : false;
+        const tooShort = min ? value.trim().length < min : false;
+        if (tooLong || tooShort) {
+          return false;
+        }
+      }
+
+      return true;
+    })
+    .every((validity) => validity);
```

Comment:
I'm not really happy with this solution, but there doesn't seem to be a way to reactively get the validation state of the `<n-form>` component.